include/deal.II/base/mpi_noncontiguous_partitioner.h

```diff
@@ -84,26 +83,61 @@ namespace Utilities
        *   functions separately and hereby overlap communication and
        *   computation.
        */
-      template <typename VectorType>
+      template <typename Number>
       void
-      update_values(VectorType &dst, const VectorType &src) const;
+      export_to_ghosted_array(
+        const ArrayView<const Number> &locally_owned_array,
+        const ArrayView<Number> &      ghost_array) const;
```

Comment:
Can you add a comment to this function (and maybe some of the functions below) how large the array is supposed to be? It should correspond to `indices_has.size()` and `indices_want.size()` if I remember correctly.